### PR TITLE
APIS-5255 - Removing email address from call to TPA

### PR DESCRIPTION
**Pull request by peteslater-ee**

### CODE REVIEW

File: app/connectors/ThirdPartyApplicationConnector.scala [hunk 1]
```diff
@@ -65,6 +66,26 @@ abstract class ThirdPartyApplicationConnector(config: ApplicationConfig, metrics
   def update(applicationId: ApplicationId, request: UpdateApplicationRequest)(implicit hc: HeaderCarrier): Future[ApplicationUpdateSuccessful] = metrics.record(api) {
     http.POST[UpdateApplicationRequest,ErrorOrUnit](s"$serviceBaseUrl/application/${applicationId.value}", request).map(throwOr(ApplicationUpdateSuccessful))
   }
+
```
Comment (AndySpaven) [inline app/connectors/ThirdPartyApplicationConnector.scala:69]: Need to remove logs in mirrored "email" function as we could still log emails if called.

File: component/steps/ApplicationsSteps.scala [hunk 2]
```diff
@@ -79,8 +83,10 @@ class ApplicationsSteps extends ScalaDsl with EN with Matchers with NavigationSu
   }
 
   Given("""^I have no application assigned to my email '(.*)'$""") { (email: String) =>
-    ApplicationStub.configureUserApplications(email)
-    ApplicationStub.configureUserApplications(email)
+    val userId = idOf(email)
+
+    ApplicationStub.configureUserApplications(userId)
+    ApplicationStub.configureUserApplications(userId)
```
Comment (AndySpaven) [inline component/steps/ApplicationsSteps.scala:89]: Do we know why this is here twice ?

File: test-utils/TestApplications.scala [hunk 3]
```diff
@@ -105,7 +105,7 @@ trait TestApplications {
 
     final def withDescription(description: Option[String]): Application = app.copy(description = description)
 
-    final def withTeamMember(email: String, userRole: Role): Application = app.copy(collaborators = app.collaborators + Collaborator(email, userRole, Some(UserId.random)))
+    final def withTeamMember(email: String, userRole: Role): Application = app.copy(collaborators = app.collaborators + Collaborator(email, userRole, UserId.random))
```
Comment (AndySpaven) [inline test-utils/TestApplications.scala:108]: should this random id be retrieved from idOf ?

File: test-utils/UserIdTracker.scala [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+// import model.Collaborator
+// import model.CollaboratorRole._
+
+trait UserIdTracker {
+  private val idsByEmail = mutable.Map[String, UserId]()
+  
+  def idOf(email: String): UserId = idsByEmail.getOrElseUpdate(email, UserId.random)
+}
+
+// trait CollaboratorTracker extends UserIdTracker {
```
Comment (AndySpaven) [inline test-utils/UserIdTracker.scala:30]: Dead code from the copy OR Code we should be using to build collaborators

File: test/controllers/ManageTeamSpec.scala [hunk 5]
```diff
@@ -73,9 +73,9 @@ class ManageTeamSpec extends BaseControllerSpec with SubscriptionTestHelperSugar
 
     fetchSessionByIdReturns(sessionId, session)
     updateUserFlowSessionsReturnsSuccessfully(sessionId)
-    when(applicationServiceMock.addTeamMember(any[Application], any[String], any[Collaborator])(any[HeaderCarrier]))
+    when(applicationServiceMock.addTeamMember(*,*,*)(*))
       .thenReturn(successful(()))
-    when(applicationServiceMock.removeTeamMember(any[Application], any[String], eqTo(loggedInUser.email))(any[HeaderCarrier]))
+    when(applicationServiceMock.removeTeamMember(*,*, eqTo(loggedInUser.email))(*))
```
Comment (AndySpaven) [inline test/controllers/ManageTeamSpec.scala:78]: Maybe another endpoint that should be migrated?

File: app/connectors/ThirdPartyApplicationConnector.scala (hunk 1 above)
Comment (peteslater-ee) [reply app/connectors/ThirdPartyApplicationConnector.scala:69]: 👍

File: component/steps/ApplicationsSteps.scala (hunk 2 above)
Comment (peteslater-ee) [reply component/steps/ApplicationsSteps.scala:89]: No idea but it seems that we don't need both of them :)

File: test/controllers/ManageTeamSpec.scala (hunk 5 above)
Comment (AndySpaven) [reply test/controllers/ManageTeamSpec.scala:78]: On second thoughts.

File: test-utils/UserIdTracker.scala (hunk 4 above)
Comment (peteslater-ee) [reply test-utils/UserIdTracker.scala:30]: Done

File: test-utils/TestApplications.scala (hunk 3 above)
Comment (peteslater-ee) [reply test-utils/TestApplications.scala:108]: Yes

File: test-utils/TestApplications.scala [hunk 6]
```diff
@@ -129,4 +130,4 @@ trait TestApplications {
   }
 }
 
-object TestApplications extends TestApplications
+// object TestApplications extends TestApplications with CollaboratorTracker with LocalUserIdTracker
```
Comment (peteslater-ee) [inline test-utils/TestApplications.scala:133]: Remove

**Review by mattgreenuk — APPROVED** [review]

**Pull request merged**